libbeat/beat/beat.go

```diff
@@ -138,20 +141,35 @@ func init() {
 // implementation. bt is the `Creator` callback for creating a new beater
 // instance.
 func Run(name, version string, bt Creator) error {
-	return handleError(newBeat(name, version).launch(bt))
+	return handleError(func() error {
+		b, err := newBeat(name, version)
+		if err != nil {
+			return err
+		}
+		return b.launch(bt)
+	}())
 }
 
 // newBeat creates a new beat instance
-func newBeat(name, version string) *Beat {
+func newBeat(name, version string) (*Beat, error) {
 	if version == "" {
 		version = defaultBeatVersion
 	}
 
-	return &Beat{
-		Name:    name,
-		Version: version,
-		UUID:    uuid.NewV4(),
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
 	}
+
+	return &Beat{
+		Info: common.BeatInfo{
+			Beat:     name,
+			Version:  version,
+			Name:     hostname,
```

Comment:
name is the name configured via config file and defaults to hostname. beat.Beat is the beat name like `filebeat`, `packetbeat`. We set `Name` later after having parsed the config file.